pkg/services/ngalert/state/template/template.go

```diff
@@ -42,8 +43,9 @@ func (l Labels) String() string {
 // Value contains the labels and value of a Reduce, Math or Threshold
 // expression for a series.
 type Value struct {
-	Labels Labels
-	Value  float64
+	Labels   Labels
+	Value    float64
+	nodeType expr.NodeType
```

Comment:
I am not fan of importing `expr` package here. I see that we only use one value of that enum. Let's replace it with `IsDatasourceNode bool`.

Reply:
Good point, updated: https://github.com/grafana/grafana/pull/102301/commits/2fb15870e2b90d8ddf49f02798b0161b578f48ee :)